joint_limits/include/joint_limits/joint_limits.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \author Adolfo Rodriguez Tsouroukdissian
+
+#ifndef JOINT_LIMITS__JOINT_LIMITS_HPP_
+#define JOINT_LIMITS__JOINT_LIMITS_HPP_
+
+#include <limits>
+#include <sstream>
+#include <string>
+
+namespace joint_limits
+{
+/**
+ * JointLimits structure stores values from from yaml definition or `<limits>` tag in URDF.
+ * The mapping from URDF attributes to members is the following:
+ *   lower --> min_position
+ *   upper --> max_position
+ *   velocity --> max_velocity
+ *   effort --> max_effort
+ */
+struct JointLimits
+{
+  JointLimits()
+  : min_position(std::numeric_limits<double>::quiet_NaN()),
+    max_position(std::numeric_limits<double>::quiet_NaN()),
+    max_velocity(std::numeric_limits<double>::quiet_NaN()),
+    max_acceleration(std::numeric_limits<double>::quiet_NaN()),
+    max_jerk(std::numeric_limits<double>::quiet_NaN()),
+    max_effort(std::numeric_limits<double>::quiet_NaN()),
+    has_position_limits(false),
+    has_velocity_limits(false),
+    has_acceleration_limits(false),
+    has_jerk_limits(false),
+    has_effort_limits(false),
+    angle_wraparound(false)
+  {
+  }
+
+  double min_position;
+  double max_position;
+  double max_velocity;
+  double max_acceleration;
+  double max_jerk;
+  double max_effort;
+
+  bool has_position_limits;
+  bool has_velocity_limits;
+  bool has_acceleration_limits;
+  bool has_jerk_limits;
+  bool has_effort_limits;
+  bool angle_wraparound;
+
+  std::string to_string()
+  {
+    std::stringstream ss_output;
+
+    if (has_position_limits)
+    {
+      ss_output << "  position limits: "
+                << "[" << min_position << ", " << max_position << "]\n";
+    }
+    if (has_velocity_limits)
+    {
+      ss_output << "  velocity limit: "
+                << "[" << max_velocity << "]\n";
+    }
+    if (has_acceleration_limits)
+    {
+      ss_output << "  acceleration limit: "
+                << "[" << max_acceleration << "]\n";
+    }
+    if (has_jerk_limits)
+    {
+      ss_output << "  jerk limit: "
+                << "[" << max_jerk << "]\n";
+    }
+    if (has_effort_limits)
+    {
+      ss_output << "  effort limit: "
+                << "[" << max_effort << "]\n";
+    }
+    if (angle_wraparound)
+    {
+      ss_output << "  angle wraparound is active.";
+    }
+
+    return ss_output.str();
+  }
+
+  std::string debug_to_string()
```

Comment:
this seems awfully similar to `to_string()` above. Can't we merge these two somehow?